# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1533145129 14400
#      Wed Aug 01 13:38:49 2018 -0400
# Node ID fc410b574693f935733c077bc6104b1b6504a4ec
# Parent  ad0b4817c09dd874a82f6ac147a3dc1bff2d4d4c
# Parent  216321ca4cc42aea1cc421ebc3422fc144d8eac4
maint: Merge stable to default.

diff --git a/scripts/deprecated/module.mk b/scripts/deprecated/module.mk
--- a/scripts/deprecated/module.mk
+++ b/scripts/deprecated/module.mk
@@ -3,16 +3,17 @@ FCN_FILE_DIRS += scripts/deprecated
 %canon_reldir%_FCN_FILES = \
   %reldir%/chop.m \
   %reldir%/comma.m \
   %reldir%/desktop.m \
   %reldir%/isstr.m \
   %reldir%/java2mat.m \
   %reldir%/paren.m \
   %reldir%/semicolon.m \
+  %reldir%/setstr.m \
   %reldir%/tmpnam.m \
   %reldir%/toascii.m
 
 %canon_reldir%dir = $(fcnfiledir)/deprecated
 
 %canon_reldir%_DATA = $(%canon_reldir%_FCN_FILES)
 
 FCN_FILES += $(%canon_reldir%_FCN_FILES)
diff --git a/scripts/deprecated/setstr.m b/scripts/deprecated/setstr.m
new file mode 100644
--- /dev/null
+++ b/scripts/deprecated/setstr.m
@@ -0,0 +1,40 @@
+## Copyright (C) 2003-2018 John W. Eaton
+##
+## This file is part of Octave.
+##
+## Octave is free software; you can redistribute it and/or modify it
+## under the terms of the GNU General Public License as published by
+## the Free Software Foundation; either version 3 of the License, or (at
+## your option) any later version.
+##
+## Octave is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+## General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Octave; see the file COPYING.  If not, see
+## <http://www.gnu.org/licenses/>.
+
+## -*- texinfo -*-
+## @deftypefn {Function File} {} setstr (@var{s})
+## This function has been deprecated.  Use char instead.
+## @end deftypefn
+
+## Author: jwe
+
+## Deprecated in version 3.0
+## Matlab still has this function, so don't remove just yet.
+
+function retval = setstr (varargin)
+
+  persistent warned = false;
+  if (! warned)
+    warned = true;
+    warning ("Octave:deprecated-function",
+             "setstr is obsolete and will be removed from a future version of Octave; please use char instead");
+  endif
+
+  retval = char (varargin{:});
+
+endfunction
